[Bug]: `pnpm typecheck` in `apps/www` has errors

### Environment

```bash
Developement/Production OS: Windows 10 19043.1110
Node version: 16.0.0
Package manager: pnpm@8.6.0
Radix Vue version: 1.0.0
Shadcn Vue version: 1.0.0
Vue version: 3.0.0
Nuxt version: 3.0.0
Nuxt mode: universal
Nuxt target: server
CSS framework: tailwindcss@3.3.3
Client OS: Windows 10 19043.1110
Browser: Chrome 90.0.4430.212
```


### Link to minimal reproduction

In the repo

### Steps to reproduce

Download the repo
```
pnpm install
cd app/www
pnpm typeheck
```

### Describe the bug

```
> www@0.8.4 typecheck /private/tmp/shadcn-vue/apps/www
> vue-tsc --noEmit

src/examples/dashboard/components/TeamSwitcher.vue:95:18 - error TS2322: Type '(list: string[] | number[] | false[] | true[] | object[], term: string) => (string | number | boolean | object)[]' is not assignable to type '(val: string[] | number[] | false[] | true[] | object[], term: string) => string[] | number[] | false[] | true[] | object[]'.
  Type '(string | number | boolean | object)[]' is not assignable to type 'string[] | number[] | false[] | true[] | object[]'.

95         <Command :filter-function="(list, term) => list.filter(i => i.label?.toLowerCase()?.includes(term)) ">
                    ~~~~~~~~~~~~~~~~

  ../../node_modules/.pnpm/radix-vue@1.2.3_vue@3.3.7/node_modules/radix-vue/dist/Combobox/ComboboxRoot.d.ts:43:5
    43     filterFunction?: (val: ArrayOrWrapped<T>, term: string) => ArrayOrWrapped<T>;
           ~~~~~~~~~~~~~~
    The expected type comes from property 'filterFunction' which is declared here on type 'Partial<{}> & Omit<{ readonly as?: AsTag | Component | undefined; readonly defaultValue?: AcceptableValue | undefined; readonly modelValue?: AcceptableValue | undefined; ... 12 more ...; "onUpdate:searchTerm"?: ((value: string) => any) | undefined; } & ... 4 more ... & { ...; }, never> & Record<...>'

src/examples/dashboard/components/TeamSwitcher.vue:95:71 - error TS2339: Property 'label' does not exist on type 'string | number | boolean | object'.
  Property 'label' does not exist on type 'string'.

95         <Command :filter-function="(list, term) => list.filter(i => i.label?.toLowerCase()?.includes(term)) ">
                                                                         ~~~~~

src/examples/tasks/components/DataTableFacetedFilter.vue:77:9 - error TS2322: Type '(list: { label: string; value: string; icon?: Component | undefined; }[], term: string) => { label: string; value: string; icon?: Component | undefined; }[]' is not assignable to type '(val: string[] | number[] | false[] | true[] | object[], term: string) => string[] | number[] | false[] | true[] | object[]'.
  Types of parameters 'list' and 'val' are incompatible.
    Type 'string[] | number[] | false[] | true[] | object[]' is not assignable to type '{ label: string; value: string; icon?: Component | undefined; }[]'.

77         :filter-function="(list: DataTableFacetedFilter['options'], term) => list.filter(i => i.label.toLowerCase()?.includes(term)) "
           ~~~~~~~~~~~~~~~~

  ../../node_modules/.pnpm/radix-vue@1.2.3_vue@3.3.7/node_modules/radix-vue/dist/Combobox/ComboboxRoot.d.ts:43:5
    43     filterFunction?: (val: ArrayOrWrapped<T>, term: string) => ArrayOrWrapped<T>;
           ~~~~~~~~~~~~~~
    The expected type comes from property 'filterFunction' which is declared here on type 'Partial<{}> & Omit<{ readonly as?: AsTag | Component | undefined; readonly defaultValue?: AcceptableValue | undefined; readonly modelValue?: AcceptableValue | undefined; ... 12 more ...; "onUpdate:searchTerm"?: ((value: string) => any) | undefined; } & ... 4 more ... & { ...; }, never> & Record<...>'

src/lib/registry/default/example/CardChat.vue:146:9 - error TS2322: Type '(list: { name: string; email: string; avatar: string; }[], search: string) => { name: string; email: string; avatar: string; }[]' is not assignable to type '(val: string[] | number[] | false[] | true[] | object[], term: string) => string[] | number[] | false[] | true[] | object[]'.
  Types of parameters 'list' and 'val' are incompatible.
    Type 'string[] | number[] | false[] | true[] | object[]' is not assignable to type '{ name: string; email: string; avatar: string; }[]'.

146         :filter-function="(list: User[], search) => list.filter(l => l.name.toLowerCase().includes(search.toLowerCase()))"
            ~~~~~~~~~~~~~~~~

  ../../node_modules/.pnpm/radix-vue@1.2.3_vue@3.3.7/node_modules/radix-vue/dist/Combobox/ComboboxRoot.d.ts:43:5
    43     filterFunction?: (val: ArrayOrWrapped<T>, term: string) => ArrayOrWrapped<T>;
           ~~~~~~~~~~~~~~
    The expected type comes from property 'filterFunction' which is declared here on type 'Partial<{}> & Omit<{ readonly as?: AsTag | Component | undefined; readonly defaultValue?: AcceptableValue | undefined; readonly modelValue?: AcceptableValue | undefined; ... 12 more ...; "onUpdate:searchTerm"?: ((value: string) => any) | undefined; } & ... 4 more ... & { ...; }, never> & Record<...>'

src/lib/registry/default/example/ComboboxDemo.vue:49:16 - error TS2322: Type '(list: typeof frameworks, search: string) => { value: string; label: string; }[]' is not assignable to type '(val: string[] | number[] | false[] | true[] | object[], term: string) => string[] | number[] | false[] | true[] | object[]'.

49       <Command :filter-function="filterFunction">
                  ~~~~~~~~~~~~~~~~

  ../../node_modules/.pnpm/radix-vue@1.2.3_vue@3.3.7/node_modules/radix-vue/dist/Combobox/ComboboxRoot.d.ts:43:5
    43     filterFunction?: (val: ArrayOrWrapped<T>, term: string) => ArrayOrWrapped<T>;
           ~~~~~~~~~~~~~~
    The expected type comes from property 'filterFunction' which is declared here on type 'Partial<{}> & Omit<{ readonly as?: AsTag | Component | undefined; readonly defaultValue?: AcceptableValue | undefined; readonly modelValue?: AcceptableValue | undefined; ... 12 more ...; "onUpdate:searchTerm"?: ((value: string) => any) | undefined; } & ... 4 more ... & { ...; }, never> & Record<...>'

src/lib/registry/default/example/ComboboxDemo.vue:59:17 - error TS2322: Type 'AcceptableValue | undefined' is not assignable to type '{ value: string; label: string; } | undefined'.
  Type 'string' is not assignable to type '{ value: string; label: string; }'.

59                 value = ev.detail.value
                   ~~~~~

src/lib/registry/new-york/example/CardChat.vue:146:9 - error TS2322: Type '(list: { name: string; email: string; avatar: string; }[], search: string) => { name: string; email: string; avatar: string; }[]' is not assignable to type '(val: string[] | number[] | false[] | true[] | object[], term: string) => string[] | number[] | false[] | true[] | object[]'.
  Types of parameters 'list' and 'val' are incompatible.
    Type 'string[] | number[] | false[] | true[] | object[]' is not assignable to type '{ name: string; email: string; avatar: string; }[]'.

146         :filter-function="(list: User[], search) => list.filter(l => l.name.toLowerCase().includes(search.toLowerCase()))"
            ~~~~~~~~~~~~~~~~

  ../../node_modules/.pnpm/radix-vue@1.2.3_vue@3.3.7/node_modules/radix-vue/dist/Combobox/ComboboxRoot.d.ts:43:5
    43     filterFunction?: (val: ArrayOrWrapped<T>, term: string) => ArrayOrWrapped<T>;
           ~~~~~~~~~~~~~~
    The expected type comes from property 'filterFunction' which is declared here on type 'Partial<{}> & Omit<{ readonly as?: AsTag | Component | undefined; readonly defaultValue?: AcceptableValue | undefined; readonly modelValue?: AcceptableValue | undefined; ... 12 more ...; "onUpdate:searchTerm"?: ((value: string) => any) | undefined; } & ... 4 more ... & { ...; }, never> & Record<...>'

src/lib/registry/new-york/example/ComboboxDemo.vue:49:32 - error TS2322: Type '(list: typeof frameworks, search: string) => { value: string; label: string; }[]' is not assignable to type '(val: string[] | number[] | false[] | true[] | object[], term: string) => string[] | number[] | false[] | true[] | object[]'.

49       <Command v-model="value" :filter-function="filterFunction">
                                  ~~~~~~~~~~~~~~~~

  ../../node_modules/.pnpm/radix-vue@1.2.3_vue@3.3.7/node_modules/radix-vue/dist/Combobox/ComboboxRoot.d.ts:43:5
    43     filterFunction?: (val: ArrayOrWrapped<T>, term: string) => ArrayOrWrapped<T>;
           ~~~~~~~~~~~~~~
    The expected type comes from property 'filterFunction' which is declared here on type 'Partial<{}> & Omit<{ readonly as?: AsTag | Component | undefined; readonly defaultValue?: AcceptableValue | undefined; readonly modelValue?: AcceptableValue | undefined; ... 12 more ...; "onUpdate:searchTerm"?: ((value: string) => any) | undefined; } & ... 4 more ... & { ...; }, never> & Record<...>'

src/lib/registry/new-york/example/ComboboxDemo.vue:59:17 - error TS2322: Type 'AcceptableValue | undefined' is not assignable to type '{ value: string; label: string; } | undefined'.
  Type 'string' is not assignable to type '{ value: string; label: string; }'.

59                 value = ev.detail.value
                   ~~~~~


Found 9 errors in 6 files.

Errors  Files
     2  src/examples/dashboard/components/TeamSwitcher.vue:95
     1  src/examples/tasks/components/DataTableFacetedFilter.vue:77
     1  src/lib/registry/default/example/CardChat.vue:146
     2  src/lib/registry/default/example/ComboboxDemo.vue:49
     1  src/lib/registry/new-york/example/CardChat.vue:146
     2  src/lib/registry/new-york/example/ComboboxDemo.vue:49
```

### Expected behavior

Typechecking should work (otherwise I have issues when including shadcn in my own app which is itself typechecked)

### Conext & Screenshots (if applicable)

_No response_

Hmmmm, weirdly in the repo I only have typecheck issues on examples, but in my own repo (with copy of the code), I have this:

```
src/checkbox/Checkbox.vue:19:4 - error TS2345: Argument of type '{ value: Readonly<LooseRequired<CheckboxRootProps>> & { readonly disabled: boolean; readonly asChild: boolean; readonly required: boolean; readonly defaultChecked: boolean; } & Record<...>; [ComputedRefSymbol]: true; effect: ReactiveEffect<...>; [RefSymbol]: true; }' is not assignable to parameter of type 'Partial<{ value: string; as: AsTag | Component<any, any, any, ComputedOptions, MethodOptions>; checked: boolean | "indeterminate"; }> & Omit<...> & Record<...>'.
  Type '{ value: Readonly<LooseRequired<CheckboxRootProps>> & { readonly disabled: boolean; readonly asChild: boolean; readonly required: boolean; readonly defaultChecked: boolean; } & Record<...>; [ComputedRefSymbol]: true; effect: ReactiveEffect<...>; [RefSymbol]: true; }' is not assignable to type 'Partial<{ value: string; as: AsTag | Component<any, any, any, ComputedOptions, MethodOptions>; checked: boolean | "indeterminate"; }>'.
    Types of property 'value' are incompatible.
      Type 'Readonly<LooseRequired<CheckboxRootProps>> & { readonly disabled: boolean; readonly asChild: boolean; readonly required: boolean; readonly defaultChecked: boolean; } & Record<...>' is not assignable to type 'string | undefined'.

19   <CheckboxRoot
      ~~~~~~~~~~~~

src/switch/Switch.vue:19:4 - error TS2345: Argument of type '{ value: Readonly<LooseRequired<SwitchRootProps & { class?: string | undefined; }>> & { readonly disabled: boolean; readonly asChild: boolean; readonly required: boolean; readonly defaultChecked: boolean; readonly checked: boolean; } & Record<...>; [ComputedRefSymbol]: true; effect: ReactiveEffect<...>; [RefSymbol]:...' is not assignable to parameter of type 'Partial<{ value: string; as: AsTag | Component<any, any, any, ComputedOptions, MethodOptions>; checked: boolean; }> & Omit<...> & Record<...>'.
  Type '{ value: Readonly<LooseRequired<SwitchRootProps & { class?: string | undefined; }>> & { readonly disabled: boolean; readonly asChild: boolean; readonly required: boolean; readonly defaultChecked: boolean; readonly checked: boolean; } & Record<...>; [ComputedRefSymbol]: true; effect: ReactiveEffect<...>; [RefSymbol]:...' is not assignable to type 'Partial<{ value: string; as: AsTag | Component<any, any, any, ComputedOptions, MethodOptions>; checked: boolean; }>'.
    Types of property 'value' are incompatible.
      Type 'Readonly<LooseRequired<SwitchRootProps & { class?: string | undefined; }>> & { readonly disabled: boolean; readonly asChild: boolean; readonly required: boolean; readonly defaultChecked: boolean; readonly checked: boolean; } & Record<...>' is not assignable to type 'string | undefined'.

19   <SwitchRoot

```

--- a/apps/www/package.json
+++ b/apps/www/package.json
@@ -28,7 +28,7 @@
     "codesandbox": "^2.2.3",
     "date-fns": "^2.30.0",
     "lucide-vue-next": "^0.276.0",
-    "radix-vue": "^1.2.3",
+    "radix-vue": "^1.2.5",
     "tailwindcss-animate": "^1.0.7",
     "v-calendar": "^3.1.2",
     "vee-validate": "4.12.3",

--- a/apps/www/src/lib/registry/default/example/ComboboxDemo.vue
+++ b/apps/www/src/lib/registry/default/example/ComboboxDemo.vue
@@ -1,7 +1,7 @@
 <script setup lang="ts">
-import { Check, ChevronsUpDown } from 'lucide-vue-next'
-
 import { ref } from 'vue'
+import { CaretSortIcon, CheckIcon } from '@radix-icons/vue'
+
 import { cn } from '@/lib/utils'
 import { Button } from '@/lib/registry/default/ui/button'
 import {
@@ -27,9 +27,9 @@ const frameworks = [
 ]
 
 const open = ref(false)
-const value = ref<typeof frameworks[number]>()
+const value = ref<string>('')
 
-const filterFunction = (list: typeof frameworks, search: string) => list.filter(i => i.value.toLowerCase().includes(search.toLowerCase()))
+// const filterFunction = (list: typeof frameworks, search: string) => list.filter(i => i.value.toLowerCase().includes(search.toLowerCase()))
 </script>
 
 <template>
@@ -41,33 +41,36 @@ const filterFunction = (list: typeof frameworks, search: string) => list.filter(
         :aria-expanded="open"
         class="w-[200px] justify-between"
       >
-        {{ value ? value.label : 'Select framework...' }}
-        <ChevronsUpDown class="ml-2 h-4 w-4 shrink-0 opacity-50" />
+        {{ value
+          ? frameworks.find((framework) => framework.value === value)?.label
+          : "Select framework..." }}
+        <CaretSortIcon class="ml-2 h-4 w-4 shrink-0 opacity-50" />
       </Button>
     </PopoverTrigger>
     <PopoverContent class="w-[200px] p-0">
-      <Command :filter-function="filterFunction">
-        <CommandInput placeholder="Search framework..." />
+      <Command>
+        <CommandInput class="h-9" placeholder="Search framework..." />
         <CommandEmpty>No framework found.</CommandEmpty>
         <CommandList>
           <CommandGroup>
             <CommandItem
               v-for="framework in frameworks"
               :key="framework.value"
-              :value="framework"
+              :value="framework.value"
               @select="(ev) => {
-                value = ev.detail.value
-                console.log(ev)
+                if (typeof ev.detail.value === 'string') {
+                  value = ev.detail.value
+                }
                 open = false
               }"
             >
-              <Check
+              {{ framework.label }}
+              <CheckIcon
                 :class="cn(
-                  'mr-2 h-4 w-4',
-                  value?.value === framework.value ? 'opacity-100' : 'opacity-0',
+                  'ml-auto h-4 w-4',
+                  value === framework.value ? 'opacity-100' : 'opacity-0',
                 )"
               />
-              {{ framework.label }}
             </CommandItem>
           </CommandGroup>
         </CommandList>

--- a/apps/www/src/lib/registry/new-york/example/ComboboxDemo.vue
+++ b/apps/www/src/lib/registry/new-york/example/ComboboxDemo.vue
@@ -27,9 +27,9 @@ const frameworks = [
 ]
 
 const open = ref(false)
-const value = ref<typeof frameworks[number]>()
+const value = ref<string>('')
 
-const filterFunction = (list: typeof frameworks, search: string) => list.filter(i => i.value.toLowerCase().includes(search.toLowerCase()))
+// const filterFunction = (list: typeof frameworks, search: string) => list.filter(i => i.value.toLowerCase().includes(search.toLowerCase()))
 </script>
 
 <template>
@@ -41,12 +41,14 @@ const filterFunction = (list: typeof frameworks, search: string) => list.filter(
         :aria-expanded="open"
         class="w-[200px] justify-between"
       >
-        {{ value ? value.label : 'Select framework...' }}
+        {{ value
+          ? frameworks.find((framework) => framework.value === value)?.label
+          : "Select framework..." }}
         <CaretSortIcon class="ml-2 h-4 w-4 shrink-0 opacity-50" />
       </Button>
     </PopoverTrigger>
     <PopoverContent class="w-[200px] p-0">
-      <Command v-model="value" :filter-function="filterFunction">
+      <Command>
         <CommandInput class="h-9" placeholder="Search framework..." />
         <CommandEmpty>No framework found.</CommandEmpty>
         <CommandList>
@@ -54,9 +56,11 @@ const filterFunction = (list: typeof frameworks, search: string) => list.filter(
             <CommandItem
               v-for="framework in frameworks"
               :key="framework.value"
-              :value="framework"
+              :value="framework.value"
               @select="(ev) => {
-                value = ev.detail.value
+                if (typeof ev.detail.value === 'string') {
+                  value = ev.detail.value
+                }
                 open = false
               }"
             >
@@ -64,7 +68,7 @@ const filterFunction = (list: typeof frameworks, search: string) => list.filter(
               <CheckIcon
                 :class="cn(
                   'ml-auto h-4 w-4',
-                  value?.value === framework.value ? 'opacity-100' : 'opacity-0',
+                  value === framework.value ? 'opacity-100' : 'opacity-0',
                 )"
               />
             </CommandItem>

